modules/objdetect/src/qrcode.cpp

```diff
@@ -1245,5 +1228,1208 @@ cv::String QRCodeDetector::detectAndDecode(InputArray in,
+                             if (cl.computeTransformationPoints(x))
+                             {
+                                 if((cl.checkPointsInsideQuadrangle(cl.transformation_points[x], not_resized_loc_points)) == true)
+                                 {
+                                     if(cl.checkPoints(cl.transformation_points[x]))
+                                     {
+
+                                         for(int l = 0; l < 3; l++)
+                                            loc[s][all_points[s][k][l]].x = -1;
+                                         flag = true;
+                                         break;
+                                     }
+                                 }
+                              }
+
+                          }
+                      }
+                }
+                if(flag) break;
+                else
+                {
+                    cl.transformation_points[x].clear();
+                    cl.localization_points[x].clear();
+                }
+          }
+      }
+  }
+}
+
+void QRDetectMulti::init(const Mat& src, double eps_vertical_, double eps_horizontal_)
+{
+    CV_TRACE_FUNCTION();
+    CV_Assert(!src.empty());
+    const double min_side = std::min(src.size().width, src.size().height);
+    if (min_side < 512.0)
+    {
+        purpose = ZOOMING;
+        coeff_expansion = 512.0 / min_side;
+        const int width  = cvRound(src.size().width  * coeff_expansion);
+        const int height = cvRound(src.size().height  * coeff_expansion);
+        Size new_size(width, height);
+        resize(src, barcode, new_size, 0, 0, INTER_LINEAR);
+    }
+    else if (min_side > 512.0)
+    {
+        purpose = SHRINKING;
+        coeff_expansion = min_side / 512.0;
+        const int width  = cvRound(src.size().width  / coeff_expansion);
+        const int height = cvRound(src.size().height  / coeff_expansion);
+        Size new_size(width, height);
+        resize(src, barcode, new_size, 0, 0, INTER_AREA);
+    }
+    else
+    {
+        purpose = UNCHANGED;
+        coeff_expansion = 1.0;
+        barcode = src.clone();
+    }
+
+    eps_vertical   = eps_vertical_;
+    eps_horizontal = eps_horizontal_;
+    adaptiveThreshold(barcode, bin_barcode, 255, ADAPTIVE_THRESH_GAUSSIAN_C, THRESH_BINARY, 83, 2);
+    adaptiveThreshold(src, bin_barcode_fullsize, 255, ADAPTIVE_THRESH_GAUSSIAN_C, THRESH_BINARY, 83, 2);
+}
+
+vector<Point2f> QRDetectMulti::separateVerticalLines(const vector<Vec3d> &list_lines)
+{
+    CV_TRACE_FUNCTION();
+    vector<Vec3d> result;
+    int temp_length;
+    vector<Point2f> point2f_result;
+    uint8_t next_pixel;
+    vector<double> test_lines;
+    temp_length = 0;
+
+    for (size_t pnt = 0; pnt < list_lines.size(); pnt++)
+    {
+        const int x = cvRound(list_lines[pnt][0] + list_lines[pnt][2] * 0.5);
+        const int y = cvRound(list_lines[pnt][1]);
+
+        // --------------- Search vertical up-lines --------------- //
```

Comment:
There is the the similar code in `QRDetect::separateVerticalLines()`.
What is the difference?
Why is the difference?

I try to describe problem which we should avoid by unifying and reusing the same code for features finding. We must have consistent behavior between `QRDetect` and `QRDetectMulti`:
- `QRDetectMulti` - no detections ==> `QRDetect` - no detection too and vice versa
- `QRDetectMulti` - has detections ==> `QRDetect` - one of these detections (largest)
- `QRDetect` - has detection ==> `QRDetectMulti` - has detections including this one

Reply:
@alalek QRDetect::separateVerticalLines() finds the best coeff_epsilon by comparison of compactness, which kmeans returns (supposed that number of clusters is 3(1 qr-code).
If  there is more than one qr-code on image this method doesn't work correctly.
By this reason there is some changes in code in  QRDetectMulti::separateVerticalLines()
Maybe, it should be renamed to avoid the mess